llvm/lib/Target/RISCV/RISCVInstrInfo.td

```diff
@@ -2017,6 +2169,74 @@ def : Pat<(binop_allwusers<add> GPR:$rs1, (AddiPair:$rs2)),
 }
 
 //===----------------------------------------------------------------------===//
+
+// MIPS extensions
+//===----------------------------------------------------------------------===//
+
+let Predicates = [UsesMIPSCMov] in {
+def : Pat<(select (XLenVT (setne (XLenVT GPR:$rs2), (XLenVT 0))),
```

Comment:
Can we use the riscv_setne and riscv_seteq ComplexPatterns to reduce the number of variations here?

Reply:
I have tried, but have not achieved to make a pattern that matches all the combinations. An alternative can be to add new `riscv_mips_setcc` ComplexPattern I guess.